src/DocCommentParser.php

```diff
@@ -20,23 +20,88 @@ class DocCommentParser
      * getTags() getTagValues() and getDescription() methods.
      *
      * @param string $docComment A doc comment as returned by the reflection getDocComment() method
+     * @return string The parsed doc comment.
      */
     public function parseDocComment(string $docComment): string
     {
         $parsedDocComment = '';
+        $deprecatedTag = '';
         $lines = explode(chr(10), $docComment);
         foreach ($lines as $line) {
             if ($this->isDocCommentTag($line)) {
+                // Check if the line contains @deprecated, parse it
+                if (stripos($line, '@deprecated') !== false) {
+                    $deprecatedTag = $this->parseDeprecatedTag($line);
+                }
                 continue;
             }
             $parsedDocComment .= preg_replace('/\\s*\\/?[\\\\*]*(.*)$/', '$1', $line) . chr(10);
         }
-        $parsedDocComment = trim($parsedDocComment);
+        $parsedDocComment = trim($parsedDocComment, " \n\r\t\v\x00\/");
+
+        // Append the deprecated tag if present
+        if ($deprecatedTag) {
+            $parsedDocComment .= "\n\n" . $deprecatedTag;
+        }
+
         return $parsedDocComment;
     }
 
     private function isDocCommentTag(string $line): bool
     {
         return preg_match('/^\s*\*?\s*@/', $line) === 1;
     }
+
+    /**
+     * Parses the deprecated tag and returns the RST formatted string.
+     *
+     * @param string $line The line containing the @deprecated tag.
+     * @return string The formatted deprecated string.
+     */
+    private function parseDeprecatedTag(string $line): string
+    {
+        // Extract the version and reason from the @deprecated tag
+        if (preg_match('/@deprecated(?:.*since\s+(?:TYPO3\s+)?v?(\d+(\.\d+)?))?(?:.*removed\s+(?:in\s+)?(?:TYPO3\s+)?v?(\d+(\.\d+)?))?[\s,.;|-]*([^@]*)?/i', $line, $matches)) {
```

Comment:
I would prefer to simplify this. A quick search across the TYPO3 core shows that there is no clear structure to `@deprecated` annotations, and none is mentioned in [the documentation](https://docs.typo3.org/m/typo3/guide-contributionworkflow/main/en-us/Appendix/HowToDeprecateThings.html). So we shouldn't assume that VH deprecations will be in this structure.

I think that there should be a more structured way to define deprecations in the core (and/or Fluid standalone), but until we're there, I think that we should output the deprecation messages as-is in the documentation.